presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -573,19 +595,24 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
                             subPlan.getRoot(),
                             entry.getValue(),
                             builder.build(),
-                            Optional.empty(),
-                            ImmutableList.of()));
+                            Optional.empty()));
         }
 
         AggregationNode aggregationNode = new AggregationNode(
                 idAllocator.getNextId(),
                 subPlan.getRoot(),
-                aggregations.build(),
+                aggregations,
                 groupingSymbols,
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 groupIdSymbol);
 
+        if (aggregationNode.hasEmptyGroupingSet() && aggregationNode.hasNonEmptyGroupingSet() && !aggregationNode.getOrderBySymbols().isEmpty()) {
+            throw SemanticExceptions.notSupportedException(
+                    node,
+                    "ORDER BY in aggregate function with empty grouping set");
```

Comment:
Also. I know this might be somehow inefficient, but I think ORDER BY in aggregation should be supported by engine. There might be some error (e.g some rule [e.g: AddIntermediateAggregation] incorrectly fires on ORDER BY with aggregation), but other than that the query should finish.